### ISSUE TITLE
unable to replicate the result from PAPER

### ISSUE THREAD

**Issue by wolfvoid** [Biangbiang0321/SpotEdit#12]
the result is DIFFERENT from that in paper. i am uncertain if i mess up something.
however my result is consistent with [issue#10](https://github.com/Biangbiang0321/SpotEdit/issues/10), i hope the author can help me fix this.

| my replicate spotedit +  Qwen-Image-Edit  | Paper | Qwen-Image-Edit-2511 |
|--------|--------|--------|
| <img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/5b4baead-ddd7-49fd-b368-5e18ce95bfa4" /> |  <img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/d321d474-1346-47d3-a760-254d85e6fb27" /> | <img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/67d95373-0804-447e-b602-f6fe2824b495" /> | 

my replicate code is as below:

```python
import sys
import torch
from pathlib import Path
from diffusers import QwenImageEditPipeline
from diffusers.utils import load_image

sys.path.insert(0, str(Path(__file__).parent / 'SpotEdit'))
from Qwen_image_edit import generate, SpotEditConfig

pipe = QwenImageEditPipeline.from_pretrained(
    '/models/Qwen-Image-Edit',
    torch_dtype=torch.bfloat16
).to('cuda')

image = load_image('image_edit/SpotEdit/asset/dog.jpg')
image = image.resize((1024, 1024))

config = SpotEditConfig(threshold=0.15)

res = generate(
    pipe,
    image=image,
    prompt='add a scarf to the dog',
    config=config,
    num_inference_steps=50,
)

output_path = Path('image_edit/outputs/qwen_spotedit_dog_scarf.jpg')
output_path.parent.mkdir(exist_ok=True)
res.images[0].save(str(output_path))
print(f"Saved: {output_path}")

```

**Comment by Biangbiang0321** [comment]
Hi @wolfvoid, thank you so much for the detailed report and the repro code — I really appreciate the effort you put into testing this. Based on my own findings, this is largely a fundamental characteristic of the method interacting with the base model.

SpotEdit decides whether a region needs editing by scoring whether that region stays consistent between the source and the model's draft prediction. If the base model itself makes large-scale changes along the way — for example, in an "add a scarf" task it additionally alters the dog's appearance — then those regions will also be classified as edit regions, even though in human semantics they would count as non-edit regions. And this behavior varies a lot from seed to seed.

In my testing, 2509 improves considerably on this kind of drift, 2511 sometimes introduces stylization changes, and FLUX.2 [klein] also shows some stylization changes. Please note that 2509/2511 run on `QwenImageEditPlusPipeline` and pair with the `Qwen_image_edit_plus` module (not `Qwen_image_edit`). I also found that the `velocity` token write-back mode (`reuse_mode="velocity"`, now the default) helps reduce the gap between edited and non-edited regions.

If you'd like, you could also give the newly released version a try, together with the k-means clustering judge (`judge_method="LPIPS_kmeans"`, now the default for the Qwen family) — I think you may find the results more interesting.

Hope this helps, and thanks again for your interest in the project!
